contracts/TokenMigration.aes

```diff
@@ -14,46 +19,51 @@ payable contract TokenMigration =
           migrations_count = count,
           map_transfered_accounts = {} }
 
-    // 'eth_addr_str' should be passed as uppercase
-    payable stateful entrypoint migrate(amount_of_tokens: int, ae_address: address, leaf_index: int, siblings: list(string), eth_addr_str: string, eth_address: bytes(20), sig: bytes(65), msg_hash: hash) =
 
-        require(!Map.member(eth_addr_str, state.map_transfered_accounts), "This account has already transferred its tokens")
+    /*
+    token_amount - the amount of tokens that the user will migrate
+    ae_address - aeternity account address to whom the tokens will be transferred
+    leaf_index - the index of the leaf in the merkle tree, needed to calculate the root_hash
+    siblings - the siblings of the leaf, needed to calculate the root_hash,
+    sig - the eth signature, which should contain the aeternity address as a message. 
+    */
+    stateful entrypoint migrate(token_amount: int, ae_address: address, leaf_index: int, siblings: list(string), sig: bytes(65)) =
         require(Address.is_payable(ae_address), "Passed AE address is not payable")
 
-        // should successfully generate same merkle tree root hash
-        let is_contained = contained_in_merkle_tree(eth_addr_str, amount_of_tokens, leaf_index, siblings)
-        require(is_contained, "From provided data, cannot be generated same root")
+        let ae_address_str = Address.to_str(ae_address)
```

Comment:
Note: This requires FATE to do the right thing - but the plan is to use FATE so OK.

(Background info: On AEVM Address.to_str returns the just the Base58 encoding of the address, not the Base58Check encoding with prefix `ak_`...)